From 4f3a2ae13ad565eb8715957928be227ff7f9117e Mon Sep 17 00:00:00 2001
From: Kirill Kapranov <kirill.kapranov@compulab.co.il>
Date: Sat, 9 Feb 2019 21:40:45 +0200
Subject: [PATCH 06/19] sb-ucm-imx8m-rev2.0: dts: Enable I2C GPIO expander

Enable PCA9555 @i2c1 GPIO expander. i2c1 bus descriptor has been moved to
baseboard include file.

Signed-off-by: Kirill Kapranov <kirill.kapranov@compulab.co.il>
---
 arch/arm64/boot/dts/compulab/sb-ucm-imx8-rev2.dtsi | 29 ++++++++++++++++++++++
 arch/arm64/boot/dts/compulab/ucm-imx8m-mini.dts    | 23 +----------------
 2 files changed, 30 insertions(+), 22 deletions(-)
 create mode 100644 arch/arm64/boot/dts/compulab/sb-ucm-imx8-rev2.dtsi

diff --git a/arch/arm64/boot/dts/compulab/sb-ucm-imx8-rev2.dtsi b/arch/arm64/boot/dts/compulab/sb-ucm-imx8-rev2.dtsi
new file mode 100644
index 0000000..d413d8d
--- /dev/null
+++ b/arch/arm64/boot/dts/compulab/sb-ucm-imx8-rev2.dtsi
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2018 CompuLab Ltd.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ */
+
+#include <dt-bindings/pinctrl/pins-imx8mm.h>
+
+&i2c1 {
+	clock-frequency = <100000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c1>;
+	status = "okay";
+
+	pca9555:pca9555@20 {
+		compatible = "nxp,pca9555";
+		gpio-controller;
+		#gpio-cells = <2>;
+		reg = <0x20>;
+	};
+};
diff --git a/arch/arm64/boot/dts/compulab/ucm-imx8m-mini.dts b/arch/arm64/boot/dts/compulab/ucm-imx8m-mini.dts
index 64173c7..b9a3ef8 100644
--- a/arch/arm64/boot/dts/compulab/ucm-imx8m-mini.dts
+++ b/arch/arm64/boot/dts/compulab/ucm-imx8m-mini.dts
@@ -15,6 +15,7 @@
 /dts-v1/;
 
 #include "../freescale/fsl-imx8mm.dtsi"
+#include "sb-ucm-imx8-rev2.dtsi"
 
 / {
 	model = "CompuLab UCM i.MX8MM board";
@@ -250,28 +251,6 @@
 	};
 };
 
-&i2c1 {
-	clock-frequency = <400000>;
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_i2c1>;
-	status = "okay";
-
-	pca9555:pca9555@20 {
-		compatible = "nxp,pca9555";
-		/* Role #0 */
-		gpio-controller;
-		#gpio-cells = <2>;
-		/* Bus address */
-		reg = <0x20>;
-	};
-
-	eeprom@54 {
-		compatible = "atmel,24c08";
-		reg = <0x54>;
-		pagesize = <16>;
-	};
-};
-
 &i2c2 {
 	clock-frequency = <400000>;
 	pinctrl-names = "default";
-- 
1.9.1

